src/ragas/metrics/_aspect_critic.py

```diff
@@ -29,8 +29,21 @@ class AspectCriticOutput(BaseModel):
 
 
 class AspectCriticInput(BaseModel):
-    user_input: str = Field(description="The input to the model")
-    response: str = Field(description="The response from the model")
+    user_input: t.Optional[str] = Field(
+        description="The input to the llm system", default=None
+    )
+    response: t.Optional[str] = Field(
+        description="The response from the llm system", default=None
+    )
+    retrieved_contexts: t.Optional[t.List[str]] = Field(
+        description="The retrieved contexts from the llm system", default=None
+    )
+    reference_contexts: t.Optional[t.List[str]] = Field(
+        description="The reference contexts for the evaluation", default=None
+    )
+    reference: t.Optional[str] = Field(
+        description="The reference answer for evaluation", default=None
+    )
     criteria: str = Field(description="The criteria to evaluate the response")
```

Comment:
We discussed that both here and with simple criteria this should be removed. If not with this PR, I can make another PR after this is merged.